Fixed minor typos with the Vault integration (#8714)
solo-io/gloo: docs/content/guides/integrations/vault/aws-irsa-auth/_index.md

## Gloo Edge

Lastly, install Gloo Edge by using a configuration that allows Vault and IRSA credential fetching.

### Step 1: Prepare Helm overrides

Override the default settings to use Vault as a source for managing secrets. To allow for IRSA, add the `eks.amazonaws.com/role-arn` annotations, which reference the roles to assume, to the `gloo` and `discovery` service accounts.

> Both `pathPrefix` and `rootKey` will need to be adjusted when mounting on a custom KV secret engine.

Reviewer: ```suggestion
Note that you must adjust both the `pathPrefix` and `rootKey` options when you use a custom `kv` secrets engine.
```

helm install gloo gloo/gloo --namespace gloo-system --create-namespace --version $EDGE_VERSION --values helm-overrides.yaml
```

## Troubleshooting

### Access denied due to identity-based policies – implicit denial

> When registering the role in Vault with `vault write auth/aws/role/<role name>` you might encounter the following error due to insufficient action with the identity-based policy.

Reviewer: ```suggestion
When you register the role in Vault by running `vault write auth/aws/role/<role name>`, you might encounter the following error due to insufficient action with the identity-based policy.
```

```
Error writing data to auth/aws/role/dev-role-iam: Error making API request.

URL: PUT http://localhost:8200/v1/auth/aws/role/dev-role-iam
Code: 400. Errors:

* unable to resolve ARN "arn:aws:iam::account-id:role/dev-role-iam" to internal ID: AccessDenied: User: arn:aws:sts::account-id:assumed-role/foo-role/bar is not authorized to perform: iam:GetRole on resource: role dev-role-iam because no identity-based policy allows the iam:GetRole action
	status code: 403, request id: e348ee87-6d44-493b-8763-14fff6aea689
```

> To fix, add the `iam:GetRole` action to a policy attached to the assumed-role identity. In the above example to the identity `arn:aws:sts::account-id:assumed-role/foo-role/bar`.

Reviewer 2: What are the circumstances that this would/not happen? Should the GetRole action be applied to our default policy in this demo or will not everyone run into this?

Author: @sam-heilbron this is more related to the assumed role attached to EKS which Vault uses to get a token (after initiating a STS request) so that it can attach the role `auth/aws/role/<some role>`.
I believe this would also be the case in EC2.

An example of an identity this is failing on would be `AccessDenied: User: arn:aws:sts::941622571599:assumed-role/kasunt-gp-a9f0dfd21386-1-worker-role/i-0e7308814cc5d8041` which is the actual instance where the STS request originates from.

> To fix, add the `iam:GetRole` action to a policy attached to the assumed-role identity. In the above example to the identity `arn:aws:sts::account-id:assumed-role/foo-role/bar`.

Reviewer: ```suggestion
To resolve this issue, add the `iam:GetRole` action to a policy attached to the assumed-role identity. In the previous example, you would add the `iam:GetRole` action to the identity `arn:aws:sts::account-id:assumed-role/foo-role/bar`.
```
